samples/basic/threads/src/main.c

```diff
@@ -0,0 +1,60 @@
+#include <zephyr.h>
+#include <device.h>
+#include <gpio.h>
+#include <misc/printk.h>
+
+/* size of stack area used by each thread */
+#define STACKSIZE 1024
+
+/* scheduling priority used by each thread */
+#define PRIORITY 7
+
+void blink1(void)
+{
+	int cnt = 0;
+	struct device *gpioa;
+
+	gpioa = device_get_binding("GPIOA");
+	gpio_pin_configure(gpioa, 15, GPIO_DIR_OUT);
+	while (1) {
+		gpio_pin_write(gpioa, 15, (cnt + 1) % 2);
+		k_sleep(100);
+		cnt++;
+	}
+}
+
+void blink2(void)
+{
+	int cnt = 0;
+	struct device *gpiod;
+
+	gpiod = device_get_binding("GPIOD");
+	gpio_pin_configure(gpiod, 2, GPIO_DIR_OUT);
+	while (1) {
+		gpio_pin_write(gpiod, 2, cnt % 2);
+		k_sleep(1000);
+		cnt++;
+	}
+}
+
+void uart_out(void)
+{
+	int cnt = 1;
+
+	while (1) {
+		printk("Toggle USR1 LED: Counter = %d\n", cnt);
+		if (cnt >= 10) {
+			printk("Toggle USR2 LED: Counter = %d\n", cnt);
+			cnt = 0;
```

Comment:
should this be ``cnt =1;`` to match what it was initialized to outside the loop?

Reply:
there is an increment for cnt right before the loop ends, If I have cnt=1 here the next loop starts with cnt=2

Reply:
ah, ok.